static/src/js/zustand/types.ts

```diff
@@ -163,6 +163,54 @@ export type MapSlice = {
   }
 }
 
+export type PreferencesState = {
+  /** The state of the panels */
+  panelState: string
+  /** The view mode for collection lists */
+  collectionListView: string
+  /** The view mode for granule lists */
+  granuleListView: string
+  /** The sort preference for collections */
+  collectionSort: string
+  /** The sort preference for granules */
+  granuleSort: string
+  /** The map view preferences */
+  mapView: {
+    /** The zoom level of the map */
+    zoom: number
+    /** The latitude center of the map */
+    latitude: number
+    /** The base layer identifier for the map */
+    baseLayer: string
+    /** The longitude center of the map */
+    longitude: number
+    /** The projection code for the map */
+    projection: string
+    /** Array of overlay layer identifiers */
+    overlayLayers: string[]
+    /** The rotation of the map in degrees */
+    rotation: number
+  }
+  /** Flag indicating if preferences are currently being submitted */
+  isSubmitting: boolean
+  /** Flag indicating if preferences have been submitted */
+  isSubmitted: boolean
+}
+
+export type PreferencesSlice = {
+  /** The Preferences Slice of the store */
+  preferences: PreferencesState & {
```

Comment:
In redux we had this 
```
preferences: {
  preferences: {} // all the values
  isSubmitting
  isSubmitted
}
```

but this has moved all of the preferences values into that top level with the submitting flags, and all the functions. If we keep the old structure of nesting all of the preferences values I think that makes the selectors you are using in components much easier
```
const preferences = useEdscStore((state) => state.preferences.preferences)
```